src/DIRAC/Core/Security/TokenFile.py

```diff
@@ -0,0 +1,99 @@
+""" Collection of utilities for dealing with security files (i.e. token files)
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+__RCSID__ = "$Id$"
+
+import os
+import json
+import stat
+import tempfile
+
+from DIRAC import S_OK, S_ERROR
+from DIRAC.Core.Utilities import DErrno
+from DIRAC.Core.Security.Locations import getTokenLocation
+
+
+def readTokenFromFile(fileName=None):
+  """ Read token from a file
+
+      :param str fileName: filename to read
+
+      :return: S_OK(dict)/S_ERROR()
+  """
+  if not fileName:
+    fileName = getTokenLocation() or os.environ.get('DIRAC_TOKEN_FILE', "/tmp/JWTup_u%d" % os.getuid())
```

Comment:
`getTokenLocation` already checks `DIRAC_TOKEN_FILE` so you shouldn't have to do it

Reply:
Except it currently returns `False` if the file doesn't exist. It should just return the path always.

Reply:
that is, `getTokenLocation` must have a different logic than `getProxyLocation`, right?

Reply:
I don't think there is a need to make the new token stuff follow exactly how the proxies worked. If anything it's a good chance to tidy things up without needing to worry about backwards compatibility.

Reply:
fixed